### PR TITLE
Add options for OF, NVDEC and NVML support 

### PR DESCRIPTION
**Pull request by klecki**
• Options for OF, NVDEC, NVML added.
• Add ability to propagate FEATURE_ENABLED flags based on BUILD_FEATURE options. (CMAKE function)
• Adjusted the collect_headers to use absolute paths

### CODE REVIEW

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
!build

**Comment by klecki** [comment]
!build

File: Dockerfile [hunk 1]
```diff
@@ -51,28 +51,49 @@ ARG NVIDIA_DALI_BUILD_FLAVOR
 ARG GIT_SHA
 ARG DALI_TIMESTAMP
 
+# Optional build arguments
+
+ARG BUILD_TEST=ON
+ARG BUILD_BENCHMARK=ON
+# ARG BUILD_NVTX=ON
```
Comment (mzient) [inline Dockerfile]: Please add this one (may be off by default) and propagate it to BUILD_NVTX.

Comment (klecki) [reply Dockerfile]: Done.

**Pull request merged**